docs/apis/role.yaml

```diff
@@ -0,0 +1,816 @@
+              example: "Internal server error"
+
+  /roles/{id}:
+    get:
+      tags:
+        - roles
+      summary: Get a role by id
+      parameters:
+        - in: path
+          name: id
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        "200":
+          description: Role details
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Role'
+              example:
+                id: "3fa85f64-5717-4562-b3fc-2c963f66afa6"
+                name: "front-desk-agent"
+                description: "Front desk agent role with booking and customer management permissions"
+                organizationUnitId: "a839f4bd-39dc-4eaa-b5cc-210d8ecaee87"
+                permissions:
+                  - "create_reservation"
+                  - "view_reservation"
+                  - "process_payment"
+                  - "view_customer_profile"
+        "400":
+          description: Bad request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                code: "ROL-1002"
+                message: "Invalid request format"
+                description: "Role ID is required"
+        "404":
+          description: Role not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                code: "ROL-1003"
+                message: "Role not found"
+                description: "The role with the specified id does not exist"
+        "500":
+          description: Internal server error
+          content:
+            text/plain:
+              schema:
+                type: string
+              example: "Internal server error"
+
+    put:
+      tags:
+        - roles
+      summary: Update a role by id
+      parameters:
+        - in: path
+          name: id
+          required: true
+          schema:
+            type: string
+            format: uuid
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/UpdateRoleRequest'
+            example:
+              name: "senior-front-desk-agent"
+              description: "Senior front desk agent with additional permissions"
+              organizationUnitId: "a839f4bd-39dc-4eaa-b5cc-210d8ecaee87"
+              permissions:
+                - "create_reservation"
+                - "view_reservation"
+                - "update_reservation"
+                - "cancel_reservation"
+                - "process_payment"
+                - "view_payment"
+                - "refund_payment"
+                - "view_customer_profile"
+                - "update_customer_profile"
+                - "create_customer_profile"
+              assignments:
```

Comment:
Should we allow updating assignments from this endpoint, if we're not returning assignments in the `/roles/{id}` GET request?